05.ls/ls.rb

```diff
@@ -102,40 +95,43 @@ def generate_mode_characters(stat_raw)
   { file_type: file_type, owner_parmition: owner_parmition, group_parmitio: group_parmition, other_parmition: other_parmition }
 end
 
-def calculate_total_block_size(long_format_files)
-  total_block_size = 0
-  path = generate_path
-  long_format_files.each do |file|
-    file_path = "#{path}#{file[:name]}"
-    total_block_size += file[:block] if File.file?(file_path) && !File.symlink?(file_path)
+def calculate_total_block_size(file_stats)
+  base_directory = generate_base_directory
+  file_stats.sum do |file|
+    file_path = generate_file_path(base_directory, file[:name])
+    File.file?(file_path) && !File.symlink?(file_path) ? file[:block] : 0
   end
-  total_block_size
 end
 
-def l_option_add_padding(long_format_files)
-  nlink_padding, uid_paddinng, gid_paddinng = calculate_padding_size(long_format_files)
-  formated_files = []
-  long_format_files.each do |file|
-    stat = {}
-    stat[:mode] = file[:mode].to_s.ljust(10)
-    stat[:nlink] = file[:nlink].to_s.rjust(nlink_padding)
-    stat[:uid] = file[:uid].to_s.rjust(uid_paddinng)
-    stat[:gid] = file[:gid].to_s.rjust(gid_paddinng)
-    stat[:size] = file[:size].to_s.rjust(6)
-    stat[:mtime] = file[:mtime].mtime.strftime(' %_m %e %H:%M').to_s
-    stat[:name] = " #{file[:name]}"
-    stat[:symlink] = " -> #{file[:symlink]}" if file[:symlink]
-    formated_files << stat
-  end
+def generate_base_directory
+  ARGV[0] || './'
+end
 
-  formated_files
+def generate_file_path(base_directory, file_name)
+  "#{base_directory}#{file_name}"
 end
 
-def calculate_padding_size(long_format_files)
+def l_option_output(total_block_size, file_stats)
+  nlink_padding, uid_paddinng, gid_paddinng = calculate_padding_size(file_stats)
+  puts "total #{total_block_size}"
+  file_stats.each do |file|
+    print file[:mode].to_s.ljust(10)
+    print file[:nlink].to_s.rjust(nlink_padding)
+    print file[:uid].to_s.rjust(uid_paddinng)
+    print file[:gid].to_s.rjust(gid_paddinng)
+    print file[:size].to_s.rjust(6)
+    print file[:mtime].mtime.strftime(' %_m %e %H:%M').to_s
+    print " #{file[:name]}"
+    print " -> #{file[:symlink]}" if file[:symlink]
+    puts ''
```

Comment:
```suggestion
    puts
```

だけでもいいかな、と思ったんですが、これは好みの問題かもしれません。（もしくはこう書かないとrubocopに怒られる？）

Reply:
怒られないみたいなので、`''`は省略したいと思います！